Automattic/wp-calypso
# Themes: Add themes detail page

Implements part of #2875 & #2876 

This PR deals with setting up a `/themes/[x]` route, which will eventually contain details about the theme. We're only concerned with showing a fake, static title, real data comes later.

Notes:
- Due to the way routing and layout currently works, i.e. two layout trees, not isomorphic, there are some hacks here to get SSR working properly. I've inline commented them.
- `isFullScreen` layout prop introduced, so we can get rid of borders/margins/padding.

TODO:
- [ ] Find a better name for `isFullScreen`
- [x] Reduce crazy shrinkwrap diff
- [x] Factor out the `renderToString` business

To test:
- Try `/design` and click on a theme's menu, and choose details
- Try logged in and logged out
- Make sure navigation works as intended

Current UI state:
<img width="1271" alt="screen shot 2016-02-15 at 16 43 06" src="https://cloud.githubusercontent.com/assets/215074/13054991/a14ff018-d404-11e5-8025-6d7cb2f92b9e.png">
<img width="431" alt="screen shot 2016-02-15 at 16 46 02" src="https://cloud.githubusercontent.com/assets/215074/13054994/a2d28b30-d404-11e5-8b97-91c22e1e9f7b.png">
